File: api/functionapp/user/src/main/java/com/example/repository/UserRepository.java
package com.example.repository;

import com.example.model.User;
import org.springframework.stereotype.Repository;

import java.util.Map;
import java.util.HashMap;
import java.util.Optional;

/**
 * In-memory repository for User entities
 */
@Repository
public class UserRepository {
    
    private final Map<String, User> users = new HashMap<>();
    
    /**
     * Save a user
     */
    public User save(User user) {
        users.put(user.getId(), user);
        return user;
    }
    
    /**
     * Find user by ID
     */
    public Optional<User> findById(String id) {
        return Optional.ofNullable(users.get(id));
    }
    
    /**
     * Find user by email
     */
    public Optional<User> findByEmail(String email) {
        return users.values().stream()
                .filter(user -> user.getEmail().equals(email))
                .findFirst();
    }
    
    /**
     * Check if user exists by email
     */
    public boolean existsByEmail(String email) {
        return users.values().stream()
                .anyMatch(user -> user.getEmail().equals(email));
    }
    
    /**
     * Get total user count
     */
    public long count() {
        return users.size();
    }
}

File: api/functionapp/user/Repositories/UserRepository.cs
using System.Collections.Concurrent;
using UserFunction.Models;

namespace UserFunction.Repositories;

public class UserRepository : IUserRepository
{
    private readonly ConcurrentDictionary<string, User> _users = new();

    public User Save(User user)
    {
        _users[user.Id] = user;
        return user;
    }

    public User? FindById(string id)
    {
        _users.TryGetValue(id, out var user);
        return user;
    }

    public User? FindByEmail(string email)
    {
        return _users.Values.FirstOrDefault(u => 
            u.Email.Equals(email, StringComparison.OrdinalIgnoreCase));
    }

    public bool ExistsByEmail(string email)
    {
        return _users.Values.Any(u => 
            u.Email.Equals(email, StringComparison.OrdinalIgnoreCase));
    }

    public long Count()
    {
        return _users.Count;
    }
}
